### DISCUSSION TITLE
[실습코드 4-6] gradle version관련 문의

### DISCUSSION THREAD

**Discussion by gunuduru** [fsoftwareengineer/todo-application#42] [질문과 답변] [answered]
안녕하세요, 책 너무나도 재밌게 읽고 있습니다.

다름이 아니라, 실습코드 4-6이 gradle version에 따라 에러가 발생하는 듯 하여

책에서 사용중인 gradle의 버전을 여쭙고자 문의드립니다.

실습코드에서 사용중인 "compile"은 gradle 7 이상의 버전에서 deprecate되었다고 합니다.
( https://github.com/jwtk/jjwt/pull/664 )

구글링해보니 implementation으로 추가하는 방식을 발견하여 문제는 해결하였으나
( https://www.inflearn.com/questions/237363 )

저처럼 최근에 책을 구입하여 별다른 gradle 버전 설정을 하지 않는 분들은 동일한 오류를 겪지 않으실까 싶습니다.

책의 실습에서 사용하는 gradle 버전을 알려주신다면 저를 비롯한 다른 분들도 이후의 실습과정에 도움이 될 것이라 예상합니다.
(Spring boot의 버전은 책의 서두에 언급해주셔서, 유용하게 사용중입니다. 감사합니다.)

좋은 책을 만들어 주셔서 감사합니다. 시간 괜찮으실 때 답변 부탁드리겠습니다.

좋은 하루 되세요!

**Comment by leeyou34** [comment]
저도 비슷한 문제를 가지고 있었는데.  감사합니다.

gradle project로 진행하면서 jwtk/jjwt를 확인해보니...


#### Gradle

```groovy
dependencies {
    implementation 'io.jsonwebtoken:jjwt-api:0.11.2'
    runtimeOnly 'io.jsonwebtoken:jjwt-impl:0.11.2',
            // Uncomment the next line if you want to use RSASSA-PSS (PS256, PS384, PS512) algorithms:
            //'org.bouncycastle:bcprov-jdk15on:1.60',
            'io.jsonwebtoken:jjwt-jackson:0.11.2' // or 'io.jsonwebtoken:jjwt-gson:0.11.2' for gson
}
```

이렇게 dependency에 추가하라고 해서 했습니다.


다만.. 위의 같이 dependency를 추가하고 나니 그림 4-8 계정생성은 잘 되는데, 4-9 로그인 요청 결과에서 자꾸 에러나네요...

혹시 4-9 로그인 요청 결과 어떻게 나오시나요?

**Reply by fsoftwareengineer** [reply]
아래의 포스트를 참고 해주시기 바랍니다.
https://github.com/fsoftwareengineer/todo-application/discussions/48

**Reply by leeyou34** [reply]
오오 감사합니당~ =D

**Comment by fsoftwareengineer** [comment, marked as answer]
Gradle 버전은 6.8.2 입니다.

**Reply by gunuduru** [reply]
감사합니다!
